modules/claims_api/spec/requests/v2/veteran_identifier_request_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Veteran Identifier Endpoint', type: :request do
+  let(:path) { '/services/benefits/v2/veteran-id' }
+  let(:headers) { { 'Authorization': 'Bearer somerandomstuff' } }
+  let(:data) { File.read(Rails.root.join('modules', 'claims_api', 'spec', 'fixtures', 'veteran_identifier.json')) }
+
+  context 'when auth header and body params are present' do
+    describe 'veteran identifier' do
+      it 'returns an id' do
+        post path, params: data, headers: headers
+        icn = JSON.parse(response.body)['id']
+        expect(icn).to eq(ClaimsApi::V2::VeteranIdentifierController::ICN_FOR_TEST_USER)
+        expect(response.status).to eq(200)
+      end
+    end
+  end
+
+  context 'when body params are not present' do
+    let(:data) { nil }
+
+    describe 'veteran identifier' do
+      it 'returns a 422 error code' do
+        post path, params: data, headers: headers
+        expect(response.status).to eq(422)
+      end
+    end
+  end
+
+  context 'when auth header is not present' do
+    let(:headers) { nil }
+
+    describe 'veteran identifier' do
+      it 'returns a 400 error code' do
```

Comment:
shouldnt this be a 401 in this case?

Reply:
If it was checking that the token was valid, then yeah it would be a 401.
But that logic is only checking that an auth header is provided.
If it's not included, that existing logic will return a 400 because it doesn't have the required header.

Reply:
> But that logic is only checking that an auth header is provided.
i get it. all im saying is i wouldnt expect a 400 from an api that i didnt provide an auth token but it was required. i would expect it to tell me 401 because im "not authorized" whether i didnt provide an auth token at all or if i provided an invalid one

Reply:
Hmmmm, this definitely feels like a grey area.
400 makes sense because you didn't give it the thing it needed to do it's job (part of which is checking auth).
401 also makes sense because if it couldn't do its job to make sure you're auth'd, then you're not auth'd.